File: solutions/java/hamming/1/src/main/java/Hamming.java
public class Hamming {
    private int hammingCount;
    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() == rightStrand.length()) {
            int count = 0;
            for (int i = 0; i < leftStrand.length(); i++) {
                if (leftStrand.charAt(i) != rightStrand.charAt(i)) {
                    count++;
                }
            }
            this.hammingCount = count;
        } else {
            throw new IllegalArgumentException("strands must be of equal length");
        }
    }

    public int getHammingDistance() {
        return this.hammingCount;
    }
}

File: solutions/python/hamming/1/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    distance = 0
    for index, letter in enumerate(strand_a):
        if letter != strand_b[index]:
            distance += 1
    return distance
